From 6f4152cb88cd387437135f42c762513ceda0901e Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Fri, 20 Jul 2018 13:12:55 +0800
Subject: [PATCH 279/706] drm: hdp: Add FW version and other information
 display

This patch adds FW version and other information display.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit 671c3aa027c0c54522b727ed476408ade941cec2)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c |  4 +++-
 drivers/gpu/drm/imx/hdp/imx-dp.c           |  5 +++++
 drivers/gpu/drm/imx/hdp/imx-hdp.c          | 10 ++++++++--
 3 files changed, 16 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c b/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c
index 716c265a9495..1f8c3766981c 100644
--- a/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c
+++ b/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c
@@ -519,8 +519,10 @@ CDN_API_STATUS CDN_API_DPTX_Set_VIC(state_struct *state, VIC_MODES vicMode,
 	val = TU_SIZE_reg * pixelClockFreq * bitsPerPixelCalc;
 	val /= NumOfLanes * rate * 8;
 
-	if (val > 64)
+	if (val > 64) {
+		printk("CDN_ERROR_NOT_SUPPORTED val: %d\n", val);
 		return CDN_ERROR_NOT_SUPPORTED;
+	}
 
 	DP_FRAMER_TU_Param = (TU_SIZE_reg << 8) + val + (1 << 15);
 
diff --git a/drivers/gpu/drm/imx/hdp/imx-dp.c b/drivers/gpu/drm/imx/hdp/imx-dp.c
index 2d4dd5ba8fc9..ace24680fafd 100644
--- a/drivers/gpu/drm/imx/hdp/imx-dp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-dp.c
@@ -18,6 +18,7 @@
 
 void dp_fw_load(state_struct *state)
 {
+	printk("Loading DP Firmware\n");
 	CDN_API_LoadFirmware(state,
 		(u8 *)mhdp_iram0_get_ptr(),
 		mhdp_iram0_get_size(),
@@ -31,6 +32,7 @@ void dp_fw_init(state_struct *state, u32 core_rate)
 	u8 echo_resp[sizeof(echo_msg) + 1];
 	int ret;
 	u8 resp;
+	u16 ver, verlib;
 
 	/* configure the clock */
 	CDN_API_SetClock(state, core_rate/1000000);
@@ -39,6 +41,9 @@ void dp_fw_init(state_struct *state, u32 core_rate)
 
 	ret = CDN_API_CheckAlive_blocking(state);
 
+	CDN_API_General_getCurVersion(state, &ver, &verlib);
+	printk("FIRMWARE VERSION: %d, LIB VERSION: %d\n", ver, verlib);
+
 	/* turn on IP activity */
 	ret = CDN_API_MainControl_blocking(state, 1, &resp);
 
diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 2a7b27e72e54..0dbe38ccb89a 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -451,8 +451,11 @@ static int imx_get_vic_index(struct drm_display_mode *mode)
 	for (i = 0; i < VIC_MODE_COUNT; i++) {
 		if (mode->hdisplay == vic_table[i][H_ACTIVE] &&
 			mode->vdisplay == vic_table[i][V_ACTIVE] &&
-			mode->clock == vic_table[i][PIXEL_FREQ_KHZ])
-			return i;
+			mode->clock == vic_table[i][PIXEL_FREQ_KHZ]) {
+				printk("VIC_MODE %d: hdisplay: %d, vdisplay: %d, clock: %d\n",
+					i, mode->hdisplay, mode->vdisplay, mode->clock);
+				return i;
+		}
 	}
 	/* Default 1080p60 */
 	printk(KERN_INFO "default vic 2\n");
@@ -858,14 +861,17 @@ static int hpd_det_worker(void *_dp)
 	for (;;) {
 		CDN_API_Get_Event(&hdp->state, &evt);
 		if (evt & 0x1) {
+			printk("Got HPD event\n");
 			/* HPD event */
 			CDN_API_DPTX_ReadEvent_blocking(&hdp->state, &eventId, &HPDevents);
 			CDN_API_DPTX_GetHpdStatus_blocking(&hdp->state, &aux_hpd);
 			if (HPDevents & 0x1) {
+				printk("HPD event: plugin\n");
 				imx_hdp_cable_plugin(hdp);
 				hdp->cable_state = true;
 				drm_kms_helper_hotplug_event(hdp->connector.dev);
 			} else if (HPDevents & 0x2) {
+				printk("HPD event: plugout\n");
 				hdp->cable_state = false;
 				imx_hdp_cable_plugout(hdp);
 				drm_kms_helper_hotplug_event(hdp->connector.dev);
-- 
2.17.1

